ocrd_network/ocrd_network/deployer.py

```diff
@@ -0,0 +1,309 @@
+                       f"detach='{detach}', remove='{remove}'")
+
+        if not self.config or not self.config.queue.address:
+            raise ValueError('Deploying RabbitMQ has failed - missing configuration.')
+
+        client = create_docker_client(self.config.queue.address, self.config.queue.username,
+                                      self.config.queue.password, self.config.queue.keypath)
+        if not ports_mapping:
+            # 5672, 5671 - used by AMQP 0-9-1 and AMQP 1.0 clients without and with TLS
+            # 15672, 15671: HTTP API clients, management UI and rabbitmq admin, without and with TLS
+            # 25672: used for internode and CLI tools communication and is allocated from
+            # a dynamic range (limited to a single port by default, computed as AMQP port + 20000)
+            ports_mapping = {
+                5672: self.config.queue.port,
+                15672: 15672,
+                25672: 25672
+            }
+        res = client.containers.run(
+            image=image,
+            detach=detach,
+            remove=remove,
+            ports=ports_mapping,
+            # The default credentials to be used by the processing workers
+            environment=[
+                f'RABBITMQ_DEFAULT_USER={self.config.queue.credentials[0]}',
+                f'RABBITMQ_DEFAULT_PASS={self.config.queue.credentials[1]}'
+            ]
+        )
+        assert res and res.id, \
+            f'Failed to start RabbitMQ docker container on host: {self.config.mongo.address}'
+        self.mq_pid = res.id
+        client.close()
+
+        # Build the RabbitMQ Server URL to return
+        rmq_host = self.config.queue.address
+        # note, integer validation is already performed
+        rmq_port = int(self.config.queue.port)
+        # the default virtual host since no field is
+        # provided in the processing server config.yml
+        rmq_vhost = '/'
+
+        self.wait_for_rabbitmq_availability(rmq_host, rmq_port, rmq_vhost,
+                                            self.config.queue.credentials[0],
+                                            self.config.queue.credentials[1])
+
+        rabbitmq_hostinfo = f'{rmq_host}:{rmq_port}{rmq_vhost}'
+        self.log.info(f'The RabbitMQ server was deployed on host: {rabbitmq_hostinfo}')
+        return rabbitmq_hostinfo
+
+    def wait_for_rabbitmq_availability(self, host: str, port: int, vhost: str, username: str,
+                                       password: str) -> None:
+        max_waiting_steps = 15
+        while max_waiting_steps > 0:
+            try:
+                dummy_publisher = RMQPublisher(host=host, port=port, vhost=vhost)
+                dummy_publisher.authenticate_and_connect(username=username, password=password)
+            except Exception:
+                max_waiting_steps -= 1
+                sleep(2)
+            else:
+                # TODO: Disconnect the dummy_publisher here before returning...
```

Comment:
Is that a TODO we should do before merging?

Reply:
Not really. It is good to have but not a must. The difference between the current version and the TODO is that the current version just disconnects due to going out of scope instead of properly sending disconnection to the RabbitMQ server.